driver-sync/src/test/functional/com/mongodb/client/ChangeStreamsTest.java

```diff
@@ -0,0 +1,280 @@
+
+    @Test
+    public void shouldPassAllOutcomes() {
+        BsonDocument result = definition.getDocument("result");
+        MongoCursor<ChangeStreamDocument<BsonDocument>> cursor = createCursor(result);
+        if (cursor == null) {
+            return;
+        }
+
+        try {
+            handleOperations();
+
+            checkStreamValues(result, cursor);
+
+        } finally {
+            cursor.close();
+        }
+
+        checkExpectations();
+
+    }
+
+    private void checkStreamValues(final BsonDocument result, final MongoCursor<ChangeStreamDocument<BsonDocument>> cursor) {
+        for (BsonValue success : result.getArray("success", new BsonArray())) {
+            BsonDocument expected = success.asDocument();
+            ChangeStreamDocument<BsonDocument> actual = cursor.next();
+
+            BsonDocument ns = expected.getDocument("ns", new BsonDocument());
+            MongoNamespace expectedNamespace = new MongoNamespace(
+                    ns.getString("db", new BsonString("db")).getValue(),
+                    ns.getString("coll", new BsonString("coll")).getValue());
+            assertEquals(expectedNamespace, actual.getNamespace());
+            assertEquals(OperationType.fromString(expected.getString("operationType").getValue()), actual.getOperationType());
+            actual.getFullDocument().remove("_id");
+
+
+            assertEquals(expected.get("fullDocument"), actual.getFullDocument());
+        }
+    }
+
+    @Nullable
+    private MongoCursor<ChangeStreamDocument<BsonDocument>> createCursor(final BsonDocument result) {
+        MongoCursor<ChangeStreamDocument<BsonDocument>> cursor;
+        try {
+            cursor = createChangeStreamCursor();
+        } catch (MongoException e) {
+            assertEquals(result.getDocument("error", new BsonDocument()).getInt32("code", new BsonInt32(-1)).getValue(), e.getCode());
+            return null;
+        }
+        return cursor;
+    }
+
+    private void checkExpectations() {
+        if (definition.getArray("expectations").size() > 0) {
+
+            String database = definition.getString("target").getValue().equals("client") ? "admin" : namespace.getDatabaseName();
+            List<CommandEvent> expectedEvents = getExpectedEvents(definition.getArray("expectations"), database, new BsonDocument());
+            List<CommandEvent> events = commandListener.getEvents();
+
+            for (int i = 0; i < expectedEvents.size(); i++) {
+                CommandEvent expectedEvent = expectedEvents.get(i);
+                CommandEvent event = events.get(i);
+                if (expectedEvent instanceof CommandStartedEvent && event instanceof CommandStartedEvent) {
+                    BsonDocument eventCommand = ((CommandStartedEvent) event).getCommand();
+                    eventCommand.remove("$clusterTime");
+                    eventCommand.remove("lsid");
+
+                    BsonDocument eventChangeStream = eventCommand.getArray("pipeline",
```

Comment:
Remove the default values, as the pipeline should always me non-empty and start with $changeStream:

                    BsonDocument eventChangeStream = eventCommand.getArray("pipeline")
                            .get(0).asDocument().getDocument("$changeStream");

Reply:
Done.